src/envoy/mixer/config.cc

```diff
@@ -110,13 +117,19 @@ void HttpMixerConfig::Load(const Json::Object& json) {
     legacy_quotas.push_back({json.getString(kQuotaName), amount});
   }
 
-  ReadTransportConfig(json, http_config.mutable_transport());
+  TransportConfig* transport_config = http_config.mutable_transport();
+  ReadLegacyTransportConfig(json, transport_config);
 
   has_v2_config = ReadV2Config(json, &http_config);
   if (has_v2_config) {
     // If v2 config is valid, clear v1 legacy_quotas.
     legacy_quotas.clear();
   }
+
+  transport_config->set_check_cluster(
```

Comment:
We should check if check_cluster is non-empty.  If it is non-empty, it is set by Pilot, we should not override it